dotnet/src/Connectors/Connectors.AI.OpenAI/Extensions/KernelOpenAISemanticFunctionExtensions.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Connectors.AI.OpenAI;
+using Microsoft.SemanticKernel.Orchestration;
+
+#pragma warning disable IDE0130
+// ReSharper disable once CheckNamespace - Using the namespace of IKernel
+namespace Microsoft.SemanticKernel;
+#pragma warning restore IDE0130
+
+/// <summary>
+/// Class for extensions methods to define semantic functions using OpenAI request settings.
+/// </summary>
+public static class KernelOpenAISemanticFunctionExtensions
+{
+    /// <summary>
+    /// Define a string-to-string semantic function, with no direct support for input context.
+    /// The function can be referenced in templates and will receive the context, but when invoked programmatically you
+    /// can only pass in a string in input and receive a string in output.
+    /// </summary>
+    /// <param name="kernel">Semantic Kernel instance</param>
+    /// <param name="promptTemplate">Plain language definition of the semantic function, using SK template language</param>
+    /// <param name="functionName">A name for the given function. The name can be referenced in templates and used by the pipeline planner.</param>
+    /// <param name="pluginName">Optional plugin name, for namespacing and avoid collisions</param>
+    /// <param name="description">Optional description, useful for the planner</param>
+    /// <param name="maxTokens">Max number of tokens to generate</param>
+    /// <param name="temperature">Temperature parameter passed to LLM</param>
+    /// <param name="topP">Top P parameter passed to LLM</param>
+    /// <param name="presencePenalty">Presence Penalty parameter passed to LLM</param>
+    /// <param name="frequencyPenalty">Frequency Penalty parameter passed to LLM</param>
+    /// <param name="stopSequences">Strings the LLM will detect to stop generating (before reaching max tokens)</param>
+    /// <param name="chatSystemPrompt">When provided will be used to set the system prompt while using Chat Completions</param>
+    /// <param name="serviceId">When provided will be used to select the AI service used</param>
+    /// <returns>A function ready to use</returns>
+    public static ISKFunction CreateSemanticFunctionForOpenAI(
```

Comment:
Consider passing the OpenAI-related settings, such as maxTokens, temperature, etc., as a single unit - 'OpenAIRequestSettings' - to simplify the method signature. It will improve readability and will allow to add new OpenAI related properties/settings without changing the method signature.

Reply:
Done